src/components/SecondaryPanes/Breakpoints.js

```diff
@@ -149,32 +139,50 @@ class Breakpoints extends Component<Props> {
           onClick={ev => ev.stopPropagation()}
         />
         <label className="breakpoint-label" title={breakpoint.text}>
-          {renderSourceLocation(breakpoint.location.source, line, column)}
+          {breakpoint.text}
         </label>
-        <CloseButton
-          handleClick={ev => this.removeBreakpoint(ev, breakpoint)}
-          tooltip={L10N.getStr("breakpoints.removeBreakpointTooltip")}
-        />
+        <div className="breakpoint-line-close">
+          <div className="breakpoint-line">
+            {getBreakpointLocation(breakpoint.location.source, line, column)}
+          </div>
+          <CloseButton
+            handleClick={ev => this.removeBreakpoint(ev, breakpoint)}
+            tooltip={L10N.getStr("breakpoints.removeBreakpointTooltip")}
+          />
+        </div>
       </div>
     );
   }
 
   render() {
     const { breakpoints } = this.props;
-    const children =
-      breakpoints.size === 0 ? (
-        <div className="pane-info">{L10N.getStr("breakpoints.none")}</div>
-      ) : (
-        sortBy(
-          [...breakpoints.valueSeq()],
-          [
-            bp => getBreakpointFilename(bp.location.source),
-            bp => bp.location.line
-          ]
-        ).map(bp => this.renderBreakpoint(bp))
+    const children = [];
+
+    if (breakpoints.size) {
+      const groupedBreakpoints = groupBy(
+        sortBy([...breakpoints.valueSeq()], bp => bp.location.line),
+        bp => getBreakpointFilename(bp.location.source)
       );
 
-    return <div className="pane breakpoints-list">{children}</div>;
+      forOwn(groupedBreakpoints, (breakpointsInFile, filename) => {
+        children.push(
+          <div className="breakpoint-heading" title={filename} key={filename}>
+            {filename}
+          </div>,
+          ...breakpointsInFile.map(bp => this.renderBreakpoint(bp))
+        );
+      });
+    }
+
+    return (
+      <div className="pane breakpoints-list">
+        {children.length ? (
```

Comment:
it might be nice to split it up and at the top do


```js
 return <div className="pane breakpoints-list">
    {this.isEmpty() ? this.renderEmptyList() : this.renderBreakpointGroups() }
 </div>
```


then below

```js
return groups.map(group => <div>{group.breakpoints.map(this.renderBreakpoint)}</div>))
```

Reply:
Updates made!